components/zcash_address/src/encoding.rs

```diff
@@ -269,6 +313,40 @@ mod tests {
         );
     }
 
+    #[test]
+    fn tex() {
+        let p2pkh_str = "t1VmmGiyjVNeCjxDZzg7vZmd99WyzVby9yC";
+        let tex_str = "tex1s2rt77ggv6q989lr49rkgzmh5slsksa9khdgte";
+
+        // Transcode P2PKH to TEX
+        let p2pkh_zaddr: ZcashAddress = p2pkh_str.parse().unwrap();
+        assert_matches!(p2pkh_zaddr.net, Network::Main);
```

Comment:
This only tests mainnet TEX addresses; add a test for testnet.

Reply:
We don't yet have test vectors or tests for _any_ testnet addresses, and while I agree that such tests would be useful we should probably approach this systematically. I'll add a test here, but it won't be robust; we should actually do proper test vectors for TEX addresses.